### PR TITLE
DTGB-282: Added image gallery and did some refactoring of template fi…

### PR DESCRIPTION
**Pull request by meirege**
…les.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the style guide CHANGELOG accordingly.
- [ ] I have updated the gent_base CHANGELOG accordingly.

### CODE REVIEW

File: gent_base.libraries.yml [hunk 1]
```diff
@@ -3,68 +3,20 @@ global_js:
   js:
     styleguide/public/styleguide/js/base-min.js: { minified: true }
 
-header:
-  version: 1.x
-  js:
-    styleguide/public/styleguide/js/header.functions-min.js: { minified: true }
-    build/js/header.binding-min.js: { minified: true }
-  dependencies:
-    - core/jquery
```
Comment (delrueba) [inline gent_base.libraries.yml:12]: :1st_place_medal: :+1:

File: source/js/gallery.binding.js [hunk 2]
```diff
@@ -1,14 +1,22 @@
 /**
  * @file
- * Gallery component binding.
+ * Javascript binding of baguettebox.min.js.
```
Comment (delrueba) [inline source/js/gallery.binding.js]: Prefer the old comment.
This is not binding of baguettebox, this is a generic binding of **a** gallery implementation.

File: templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      {% endblock %}
+  {% else %}
+    {%- for item in items %}
+      {{ block('gallery') }}
+    {% endfor -%}
+    {{ block('gallery_extras') }}
+  {% endif %}
+{% else %}
+  <ul{{ attributes.addClass(classes) }}>
+    <div{{ title_attributes.addClass(title_classes) }}>{{ label }}</div>
```
Comment (delrueba) [inline templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig]: Only allowed children of UL are zero or more LI elements.

File: templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+      {% endfor -%}
+      {% block gallery_extras %}
+          <li class="image-gallery__show-more">
+            <div class="show-more__content">
+              <span class="show-more__total">7</span>
+              <span>{{ 'Show all photos'|t }} <i class="icon-arrow-right" aria-hidden="true"></i></span>
+            </div>
+          </li>
+        </ul>
+        <span hidden="" id="image-gallery__open-gallery">{{ 'open gallery'|t }}</span>
```
Comment (delrueba) [inline templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig]: must be unique

File: templates/custom/media/media--image--teaser.html.twig [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ * Available variables:
+ * - name: Name of the media.
+ * - content: Media content.
+ *
+ * @see template_preprocess_media()
+ *
+ * @ingroup themeable
+ */
+#}
+<a{{ attributes }} href="{{ file_url(content.field_media_image.0['#item'].entity.uri.value) }}" aria-label="image caption" aria-describedby="image-gallery__open-gallery">
```
Comment (delrueba) [inline templates/custom/media/media--image--teaser.html.twig]: Aria-label should be content of figcaption. If there is no figcaption, don't add the aria-label attribute.

Aria-describedby should reference an element with a unique ID.
We could start adding some of these generic texts to the page template... They'll be used on virtually every page anyway.

File: templates/custom/media/media--image--teaser.html.twig [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  {{ title_suffix.contextual_links }}
```
Comment (delrueba) [inline templates/custom/media/media--image--teaser.html.twig]: Won't you have links in links?

File: templates/custom/media/media--image--teaser.html.twig [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  {% if content %}
```
Comment (delrueba) [inline templates/custom/media/media--image--teaser.html.twig]: If there is no content, you'll already have issues on the A element too.

**Review by delrueba — CHANGES_REQUESTED** [review]

File: gent_base.libraries.yml (hunk 1 above)
Comment (meirege) [reply gent_base.libraries.yml:12]: 😆

File: source/js/gallery.binding.js (hunk 2 above)
Comment (meirege) [reply source/js/gallery.binding.js]: Reverted this.

File: templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig (hunk 3 above)
Comment (meirege) [reply templates/custom/dg_paragraphs/image-gallery/field--paragraph--field-image--image.html.twig]: Placed those divs outside the list.

File: templates/custom/media/media--image--teaser.html.twig (hunk 6 above)
Comment (meirege) [reply templates/custom/media/media--image--teaser.html.twig]: It doesn't actually print links as far as I know.

File: templates/custom/media/media--image--teaser.html.twig (hunk 7 above)
Comment (meirege) [reply templates/custom/media/media--image--teaser.html.twig]: Mmm true.

File: templates/custom/media/media--image--teaser.html.twig (hunk 5 above)
Comment (meirege) [reply templates/custom/media/media--image--teaser.html.twig]: Agreed.

**Review by delrueba — APPROVED** [review]

**Pull request merged**